usb/usb_host_hid/include/usb/hid_host.h

```diff
@@ -0,0 +1,267 @@
+    void *callback_arg;                     /**< User provided argument passed to callback */
+} hid_host_driver_config_t;
+
+/**
+ * @brief HID device configuration structure (HID Interface)
+*/
+typedef struct {
+    hid_host_interface_event_cb_t callback;     /**< Callback invoked when HID Interface event occurs */
+    void *callback_arg;                         /**< User provided argument passed to callback */
+} hid_host_device_config_t;
+
+/**
+ * @brief USB HID Host install USB Host HID Class driver
+ *
+ * @param[in] config configuration structure HID to create
+ * @return esp_err_r
+ */
+esp_err_t hid_host_install(const hid_host_driver_config_t *config);
+
+/**
+ * @brief USB HID Host uninstall HID Class driver
+ * @return esp_err_t
+ */
+esp_err_t hid_host_uninstall(void);
+
+/**
+ * @brief USB HID Host open a device with specific device parameters
+ *
+ * @param[in] iface_handle     Pointer to an opened HID Device
+ * @param[in] config           Configuration structure HID device to open
+ * @return esp_err_t
+ */
+esp_err_t hid_host_device_open(hid_host_device_handle_t hid_dev_handle,
+                               const hid_host_device_config_t *config);
+
+/**
+ * @brief USB HID Host close device
+ *
+ * @param[in] hid_dev_handle   Device handle obtained from hid_host_open_device function
+ * @return esp_err_t
+ */
+esp_err_t hid_host_device_close(hid_host_device_handle_t hid_dev_handle);
+
+/**
+ * @brief HID Host USB event handler
+ *
+ * If HID Host install was made with create_background_task=false configuration,
+ * application needs to handle USB Host events itself.
+ * Do not used if HID host install was made with create_background_task=true configuration
+ *
+ * @param[in]  timeout_ms  Timeout in milliseconds
+ * @return esp_err_t
+ */
+esp_err_t hid_host_handle_events(uint32_t timeout_ms);
+
+/**
+ * @brief HID Device get parameters by handle.
+ *
+ * @param[in] hid_dev_handle    HID Device handle
+ *
+ * @return dev_params           Pointer to a device parameters (USB port, Interface number, SubClass and Proto)
+ */
+hid_host_dev_params_t *hid_host_device_get_params(hid_host_device_handle_t hid_dev_handle);
+
+/**
+ * @brief HID Host get device transferred data pointer by handle
+ *
+ * @param[in] hid_dev_handle    HID Device handle
+ * @param[out] data_length  Length of transferred data
+ *
+ * @return a uint8_t pointer to a data buffer
+ */
+uint8_t *hid_host_device_get_data(hid_host_device_handle_t hid_dev_handle,
+                                  size_t *data_length);
+
+// ------------------------ USB HID Host driver API ----------------------------
+
+/**
+ * @brief HID Host enable device by handle. Calls an Interface callback when the event is occured.
+ *
+ * @param[in] hid_dev_handle  HID Device handle
+ * @return esp_err_t
+ */
+esp_err_t hid_host_device_start(hid_host_device_handle_t hid_dev_handle);
```

Comment:
Yep, you are write. 
Moreover, I still questioning myself why we need two callbacks here, because one would be enough. 
Such as event loop usage. 

I guess, that I can leave that for a next version.